lib/spack/docs/packaging_guide_1.rst

```diff
@@ -0,0 +1,3342 @@
+
+.. code-block:: python
+
+   # The virtual bar package in the same prefix as foo
+
+   # bar resolves to the foo package
+   >>> spec["bar"]
+   foo@1.0%gcc@11.3.1+bar+baz arch=linux-fedora35-haswell
+   >>> spec["bar"].prefix
+   "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6"
+
+   # home defaults to the foo prefix without either a Foo.bar_home
+   # or Foo.home implementation
+   >>> spec["bar"].home
+   "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6"
+
+   # bar header in the foo prefix
+   >>> spec["bar"].headers
+   HeaderList([
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/include/bar/bar.h"
+   ])
+
+   # bar include dirs from the foo prefix
+   >>> spec["bar"].headers.directories
+   ["/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/include"]
+
+   # bar library from the foo prefix
+   >>> spec["bar"].libs
+   LibraryList([
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/lib64/libFooBar.so"
+   ])
+
+   # bar library directories from the foo prefix
+   >>> spec["bar"].libs.directories
+   ["/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/lib64"]
+
+.. code-block:: python
+
+   # The virtual baz package in a subdirectory of foo's prefix
+
+   # baz resolves to the foo package
+   >>> spec["baz"]
+   foo@1.0%gcc@11.3.1+bar+baz arch=linux-fedora35-haswell
+   >>> spec["baz"].prefix
+   "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6"
+
+   # baz_home implementation provides the subdirectory inside the foo prefix
+   >>> spec["baz"].home
+   "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/baz"
+
+   # baz headers in the baz subdirectory of the foo prefix
+   >>> spec["baz"].headers
+   HeaderList([
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/baz/include/baz/baz.h"
+   ])
+
+   # baz include directories in the baz subdirectory of the foo prefix
+   >>> spec["baz"].headers.directories
+   [
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/baz/include"
+   ]
+
+   # baz libraries in the baz subdirectory of the foo prefix
+   >>> spec["baz"].libs
+   LibraryList([
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/baz/lib/libFooBaz.so"
+   ])
+
+   # baz library directories in the baz subdirectory of the foo prefix
+   >>> spec["baz"].libs.directories
+   [
+       "/opt/spack/linux-fedora35-haswell/gcc-11.3.1/foo-1.0-ca3rczp5omy7dfzoqw4p7oc2yh3u7lt6/baz/lib"
+   ]
+
+.. _make-package-findable:
+
+----------------------------------------------------------
+Making a package discoverable with ``spack external find``
```

Comment:
I think this whole section is missing a discussion of detection from `cls.libraries` instead of `cls.executables`.